share/empty.go

```diff
@@ -2,64 +2,66 @@ package share
 
 import (
 	"bytes"
-	"context"
 	"fmt"
-
-	"github.com/ipfs/go-blockservice"
+	"sync"
 
 	"github.com/celestiaorg/celestia-app/pkg/appconsts"
 	"github.com/celestiaorg/celestia-app/pkg/da"
 	"github.com/celestiaorg/celestia-app/pkg/shares"
 	"github.com/celestiaorg/rsmt2d"
 )
 
+// EmptyRoot returns Root of the empty block EDS.
+func EmptyRoot() *Root {
+	initEmpty()
+	return emptyBlockRoot
+}
+
+// EmptyExtendedDataSquare returns the EDS of the empty block data square.
+func EmptyExtendedDataSquare() *rsmt2d.ExtendedDataSquare {
+	initEmpty()
+	return emptyBlockEDS
+}
+
+// EmptyBlockShares returns the shares of the empty block.
+func EmptyBlockShares() []Share {
+	initEmpty()
+	return emptyBlockShares
+}
+
 var (
-	emptyRoot *Root
-	emptyEDS  *rsmt2d.ExtendedDataSquare
+	emptyMu          sync.Mutex
+	emptyBlockRoot   *Root
+	emptyBlockEDS    *rsmt2d.ExtendedDataSquare
+	emptyBlockShares []Share
 )
 
-func init() {
```

Comment:
what was the reason for changing this? did something about it not work?

Reply:
https://github.com/celestiaorg/celestia-node/pull/2388#discussion_r1241912753